packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -53,37 +49,43 @@ interface TerminalCSSProperties {
 export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget {
 
     private readonly TERMINAL = 'Terminal';
-    protected readonly onTermDidClose = new Emitter<TerminalWidget>();
-    protected terminalId = -1;
     protected term: Xterm.Terminal;
     protected restored = false;
     protected closeOnDispose = true;
-    protected waitForConnection: Deferred<MessageConnection> | undefined;
 
-    @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService;
-    @inject(WebSocketConnectionProvider) protected readonly webSocketConnectionProvider: WebSocketConnectionProvider;
     @inject(TerminalWidgetOptions) options: TerminalWidgetOptions;
-    @inject(ShellTerminalServerProxy) protected readonly shellTerminalServer: ShellTerminalServerProxy;
-    @inject(TerminalWatcher) protected readonly terminalWatcher: TerminalWatcher;
     @inject(ThemeService) protected readonly themeService: ThemeService;
     @inject(ILogger) @named('terminal') protected readonly logger: ILogger;
     @inject('terminal-dom-id') public readonly id: string;
     @inject(TerminalPreferences) protected readonly preferences: TerminalPreferences;
 
+    @inject(TerminalClientFactory) terminalClientFactory: TerminalClientFactory;
+
+    private terminalClient: TerminalClient;
+
+    protected readonly _onTerminalDidClose = new Emitter<TerminalWidget>();
```

Comment:
Remove, renaming protected properties is a breaking change, please keep names there it is possible.